### PR TITLE
Deprecate ComputeScreenPos

### PR DESCRIPTION
**Pull request by eh-unity**
### Checklist for PR maker
- [ ] Have you added a backport label (if needed)? For example, the `need-backport-*` label. After you backport the PR, the label changes to `backported-*`.
- [x] Have you updated the changelog? Each package has a `CHANGELOG.md` file.
- [ ] Have you updated or added the documentation for your PR? When you add a new feature, change a property name, or change the behavior of a feature, it's best practice to include related documentation changes in the same PR. If you do add documentation, make sure to add the relevant Graphics Docs team member as a reviewer of the PR. If you are not sure which person to add, see the [Docs team contacts sheet](https://docs.google.com/spreadsheets/d/1rgUWWgwLFEHIQ3Rz-LnK6PAKmbM49DZZ9al4hvnztOo/edit#gid=1058860420).
- [ ] Have you added a graphic test for your PR (if needed)? When you add a new feature, or discover a bug that tests don't cover, please add a graphic test.

---
### Purpose of this PR
Deprecate ComputeScreenPos as it's a duplicate and badly named/documented. It confuses people.
Remove it's usage from URP.

---
### Testing status
Describe what manual/automated tests were performed for this PR
2020.2 Pass, 072 fail is expected. https://yamato.cds.internal.unity3d.com/jobs/902-Graphics/tree/universal%252Fscreenpos-deprecate/.yamato%252Fall-universal.yml%2523All_Universal_CUSTOM-REVISION/4219839/job/pipeline
But the failed job passes  in current master with 20.2.
https://yamato.cds.internal.unity3d.com/jobs/902-Graphics/tree/universal%252Fscreenpos-deprecate/.yamato%252Funiversal-win-dx11.yml%2523Universal_Win_DX11_playmode_XR_mono_Linear_CUSTOM-REVISION/4221238/job

---
### Comments to reviewers
Notes for the reviewers you have assigned.

### CODE REVIEW

File: com.unity.render-pipelines.universal/ShaderLibrary/Deprecated.hlsl [hunk 1]
```diff
@@ -22,4 +22,17 @@
 // w is used for knowing whether the object is opaque(1) or alpha blended(0)
 half4 _DrawObjectPassData;
 
+// Deprecated: A confusingly named and duplicate function that scales clipspace to unity NDC range. (-w < x(-y) < w --> 0 < xy < w)
+// Use GetVertexPositionInputs().inputNDC.xyz instead for vertex shader
+// Or a similar function in Common.hlsl, ComputeNormalizedDeviceCoordinatesWithZ()
+//
+// TODO: A similar function should be already available in SRP lib on master. Use that instead
```
Comment (phi-lira) [inline com.unity.render-pipelines.universal/ShaderLibrary/Deprecated.hlsl:29]: remove this TODO comment

**Comment by phi-lira** [comment]
Adding @rainsing for review on 2D changes.

File: TestProjects/UniversalGraphicsTest/Assets/Scenes/130_UnityMatrixIVP/WorldReconstruction.shader [hunk 2]
```diff
@@ -56,15 +58,16 @@
 					UNITY_SETUP_INSTANCE_ID(i);
 					UNITY_SETUP_STEREO_EYE_INDEX_POST_VERTEX(i);
 
-                    float2 uv = i.scrPos.xy / i.scrPos.w;
-                    float depth = SampleSceneDepth(uv);
+				    float2 uv = i.scrPos.xy;
```
[the remark this answers is not in the document]
Comment (eh-unity) [reply TestProjects/UniversalGraphicsTest/Assets/Scenes/130_UnityMatrixIVP/WorldReconstruction.shader:61]: It's a bug, it should have / w.

**Review by rainsing — APPROVED** [review]

File: com.unity.render-pipelines.universal/ShaderLibrary/Deprecated.hlsl (hunk 1 above)
Comment (eh-unity) [reply com.unity.render-pipelines.universal/ShaderLibrary/Deprecated.hlsl]: done

**Comment by eh-unity** [comment]
This is universal only change and it's passing the tests https://yamato.cds.internal.unity3d.com/job/4296819 . CUSTOM-REVISON is 20.2, which fails because of the codestyle rename.
Should be good to go after @oleks-k docs review.

**Review by phi-lira — APPROVED** [review]
LGTM

**Review by oleks-k — APPROVED** [review]
Approving assuming that you will move writing-shaders-urp-reconstruct-world-position.md and the PNG to a separate PR as we agreed (to make the review process more optimal).
Thanks for the great input!

**Comment by eh-unity** [comment]
Separated the docs into another branch: https://github.com/Unity-Technologies/Graphics/pull/2849

**Review by Verasl — APPROVED** [review]
LGTM

**Pull request merged**